content/en/docs/spin-operator/tutorials/pub-sub-with-redis.md

```diff
@@ -0,0 +1,213 @@
+- [Rancher Desktop](https://rancherdesktop.io/) to manage Kubernetes on our Desktop. ([This page](../../spin-operator/tutorials/integrating-with-rancher-desktop.md) documents integrating Rancher Desktop and SpinKube.)
+
+## Create a Kubernetes Cluster
+
+First, we create our Kubernetes cluster:
+
+```bash
+k3d cluster create wasm-cluster --image ghcr.io/spinkube/containerd-shim-spin/k3d:v0.13.1 -p "8081:80@loadbalancer" --agents 2
+```
+
+## Install CRDs for SpinKube
+
+Next, we apply the necessary Custom Resource Definitions (CRDs) to our Kubernetes cluster:
+
+```bash
+kubectl apply -f https://github.com/spinkube/spin-operator/releases/download/v0.1.0/spin-operator.crds.yaml
+kubectl apply -f https://github.com/spinkube/spin-operator/releases/download/v0.1.0/spin-operator.runtime-class.yaml
+kubectl apply -f https://github.com/spinkube/spin-operator/releases/download/v0.1.0/spin-operator.shim-executor.yaml
+```
+
+## Install Spin Operator 
+
+Then, we install Spin Operator which handles the `SpinApp` application that we are about to create:
+
+```bash
+helm install spin-operator \
+  --namespace spin-operator \
+  --create-namespace \
+  --version 0.1.0 \
+  --wait \
+  oci://ghcr.io/spinkube/charts/spin-operator
+```
+
+## Redis
+
+Let's dive in and get Redis sorted because we are going to need information about our Redis installation in our Spin application's config. We will use the following `helm` commands to get the job done:
+
+```bash
+helm repo add bitnami https://charts.bitnami.com/bitnami
+helm repo update
+helm install my-redis bitnami/redis
+```
+
+The `helm` installation process from above prints a lot of useful information to the terminal. For example, the endpoints to communicate with Redis (read/write vs read-only):
+
+```bash
+my-redis-master.default.svc.cluster.local for read/write operations (port 6379)
+my-redis-replicas.default.svc.cluster.local for read-only operations (port 6379)
+```
+
+In addition, there are pre-written commands that you can cut and paste. For example:
+
+```bash
+export REDIS_PASSWORD=$(kubectl get secret --namespace default my-redis -o jsonpath="{.data.redis-password}" | base64 -d)
+```
+
+Go ahead and run the command above to set the password. And, if need be, you can run the following command to see the actual password printed in your terminal:
+
+```bash
+echo $REDIS_PASSWORD
+```
+
+## Create a Redis Message Handler Using Rust
+
+We use Spin's convenient `redis-rust` template to scaffold our Rust-based Redis message handler:
+
+```bash
+spin new -t redis-rust redis-message-handler
+```
+
+The command above will provide the prompts for you to add the Description, Redis address and Redis channel (We use the `my-redis-master.default.svc.cluster.local` from above to help configure the Redis address, and the channel is arbitrary i.e. `channel-one`):
+
+```bash
+Description: Redis message handler using Rust
+Redis address[redis://localhost:6379]: redis://my-redis-master.default.svc.cluster.local:6379
```

Comment:
Unless the redis chart is installed with `--set auth.enabled=false`, the address will need the password included, eg `redis://:<password>@my-redis-master.default.svc.cluster.local:6379`.  Otherwise, the connection attempt on behalf of Spin will fail and the app will error/crashloop.

Reply:
Thanks Vaughn, I will add this.